pkg/apis/deviceplugin/v1/dlbdeviceplugin_webhook.go

```diff
@@ -15,73 +15,112 @@
 package v1
 
 import (
+	"context"
+	"fmt"
+
 	"k8s.io/apimachinery/pkg/runtime"
 	ctrl "sigs.k8s.io/controller-runtime"
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
-	"sigs.k8s.io/controller-runtime/pkg/webhook"
 	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
 
 	"github.com/intel/intel-device-plugins-for-kubernetes/pkg/controllers"
 )
 
-var (
-	// dlbdevicepluginlog is for logging in this package.
-	dlbdevicepluginlog = logf.Log.WithName("dlbdeviceplugin-resource")
-
-	dlbMinVersion = controllers.ImageMinVersion
-)
+var dlbMinVersion = controllers.ImageMinVersion
 
 // SetupWebhookWithManager sets up a webhook for DlbDevicePlugin custom resources.
 func (r *DlbDevicePlugin) SetupWebhookWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewWebhookManagedBy(mgr).
 		For(r).
+		WithDefaulter(&dlbDevicePluginDefaulter{}).
+		WithValidator(&dlbDevicePluginValidator{}).
 		Complete()
 }
 
 // +kubebuilder:webhook:path=/mutate-deviceplugin-intel-com-v1-dlbdeviceplugin,mutating=true,failurePolicy=fail,groups=deviceplugin.intel.com,resources=dlbdeviceplugins,verbs=create;update,versions=v1,name=mdlbdeviceplugin.kb.io,sideEffects=None,admissionReviewVersions=v1
 
-var _ webhook.Defaulter = &DlbDevicePlugin{}
+type dlbDevicePluginDefaulter struct{}
 
-// Default implements webhook.Defaulter so a webhook will be registered for the type.
-func (r *DlbDevicePlugin) Default() {
-	dlbdevicepluginlog.Info("default", "name", r.Name)
+var _ admission.CustomDefaulter = &dlbDevicePluginDefaulter{}
 
-	if len(r.Spec.Image) == 0 {
-		r.Spec.Image = "intel/intel-dlb-plugin:" + dlbMinVersion.String()
+// Default implements admission.CustomDefaulter so a webhook will be registered for the type.
+func (r *dlbDevicePluginDefaulter) Default(ctx context.Context, obj runtime.Object) error {
+	log := logf.FromContext(ctx).WithName("dlbdeviceplugin-resource")
```

Comment:
Please use per-plugin constant for the log context name in all of these calls, instead of a free-form string.

Reply:
I might end up dropping `WithName()`s completely and replace them with a type print below.

Reply:
pushed an update to sgxdeviceplugin for reference

Reply:
Seems OK...